src/Compiler/Common.idr

```diff
@@ -332,19 +289,19 @@ getCompileData doLazyAnnots phase_in tm_in
          defs <- get Ctxt
          whenJust (dumpcases sopts) $ \ f =>
             do coreLift $ putStrLn $ "Dumping case trees to " ++ f
-               dumpCases f namedDefs
+               dumpIR {def=NamedDef} f (map (\(n, _, def) => (n, def)) namedDefs)
```

Comment:
If you're going to need to take `def` explicitly every time, I would just make it an explicit argument:

```
dumpIR : (0 def : Type) -> Show def => String -> List (Name, def) -> Core ()
```

Reply:
Thank you for your comment. Actually I don't need to take `def` explicitly. I was thinking (not too much, to be honest), that maybe it makes it more readable. I added a commit which removes the explicit passing of `dev`.